lsystems/svg.py

```diff
@@ -2,32 +2,28 @@
 from math import sin, cos, radians, tan, atan, pi
 
 
-class PathSegment:
-    """
-    represents a segment of a svg path
-    the start position and start direction (in degree) can be seen as assumptions,
-    which should be met by the previous segment to assure the end position and
-    end direction. The d contains the path segment as string representation as
-    it is required by the svg path tag. The coordinates in d should be absolute.
-    """
-
-    def __init__(self,
-                 start_position: tuple[float, float],
-                 start_direction: float,
-                 end_position: float,
-                 end_direction: float,
-                 d: str
-                 ):
-        self.start_position = start_position
-        self.start_direction = start_direction
-        self.end_position = end_position
-        self.end_direction = end_direction
-        self.d = d
+class Cursor:
+    x: float
+    y: float
+    dir: float
+
+    def __init__(self, x: float, y: float, dir: float):
+        self.x = x
+        self.y = y
+        self.dir = dir
+
+    def __eq__(self, other):
```

Comment:
perhaps we {c|sh}ould use math.isclose?
https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python

Reply:
done